packages/block-editor/src/store/selectors.js

```diff
@@ -1798,12 +1798,47 @@ export const __experimentalGetParsedPattern = createSelector(
 		if ( ! pattern ) {
 			return null;
 		}
+
+		const blocks = parse( pattern.content );
+		const { allowedBlockTypes } = getSettings( state );
+		let isAllowed = isBoolean( allowedBlockTypes )
```

Comment:
I think it was more clear before. I think you should just declare the previous IIFE into a separate function and call it - that was my intent for the IIFE comment.

Also now it will loop through blocks even if `allowedBlockTypes` is boolean and `true`.